### ISSUE TITLE
Screen not refreshing until next key press

### ISSUE THREAD

**Issue by GoogleCodeExporter** [Valyrox/lanterna#33]
```
I've got a program that reads data from network in a separate thread, then adds 
a row to a Table using GuiScreen.runInEventThread. I'd expect that being a 
separate thread it wouldn't have to wait for keyboard input to refresh, but the 
new row doesn't appear until I press a key (any key). Maybe the key reading is 
happening in the wrong (same) thread?
```

Original issue reported on code.google.com by `bem...@gmail.com` on 30 Aug 2012 at 6:52

**Comment by GoogleCodeExporter** [comment]
```
Ah, no, the table isn't telling the GUI system to repaint when you add rows. 
I'll fix it.
```

Original comment by `mab...@gmail.com` on 30 Aug 2012 at 6:54
- Changed state: **Accepted**

**Comment by GoogleCodeExporter** [comment]
```
I've fixed it on 2.0.4-SNAPSHOT and 2.1.0-SNAPSHOT and deployed to Sonatype; 
there's a test class com.googlecode.lanterna.test.issue.Issue33 which you can 
try out to verify it's working.
```

Original comment by `mab...@gmail.com` on 1 Sep 2012 at 3:43
- Changed state: **Fixed**
